master/internal/stream/projects.go

```diff
@@ -173,8 +174,8 @@ func ProjectCollectStartupMsgs(
 		return nil, err
 	}
 
-	// this really should be a union
 	exist = append(exist, newEntities...)
+	slices.Sort(exist)
```

Comment:
don't you already have an orderby id in the query?

I'm no sql expert, but I think it's normally considered good form to sort in the query rather than in memory.  I couldn't honestly say why it's important though, but I would say it's weird to sort twice.